backend/app/services/langgraph_chat/agents/graph_builder.py

```
@@ -0,0 +1,303 @@
+    def _convert_tools_to_langchain(self) -> List[Tool]:
+        """Convert tool registry to LangChain Tool objects.
+
+        Returns:
+            List of LangChain Tool instances
+        """
+        langchain_tools = []
+
+        for tool in self.tool_registry.get_all_tools():
+            # Create async function wrapper for tool execution
+            async def tool_func(**kwargs):
+                result = await tool.execute(**kwargs)
```

Ruff (0.14.8)

Function definition does not bind loop variable `tool`

(B023)